src/NuGet.Core/NuGet.CommandLine.XPlat/ReportRenderers/ConsoleRenderer/ListPackageConsoleRenderer.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.CommandLine.XPlat
+{
+    /// <summary>
+    /// Console output renderer for dotnet list package command
+    /// </summary>
+    internal class ListPackageConsoleRenderer : IReportRenderer
+    {
+        protected List<ReportProblem> _problems = new();
+
+        private ListPackageConsoleRenderer()
+        { }
+
+        public static ListPackageConsoleRenderer GetInstance()
+        {
+            return new ListPackageConsoleRenderer();
+        }
+
+        public void AddProblem(string errorText, ProblemType problemType)
+        {
+            _problems.Add(new ReportProblem(string.Empty, errorText, problemType));
+        }
+
+        public IEnumerable<ReportProblem> GetProblems()
+        {
+            return _problems;
+        }
+
+        public void AddProjectReport(ListPackageReportModel listPackageReportModel)
+        {
+            ListPackageConsoleWriter.Render(new ListPackageOutputContentV1()
```

Comment:
IMO, it's a bit misleading that `AddProjectReport` actually renders. 
I think you should have to call `Render` for the rendering to happen. 

I know this might affect when an output happens in list package when run on a solution. Do we think that's super important?

Reply:
Or another way of saying, if this renders too, then name it as such.

Reply:
Sure, renamed to just `Render`.